fw/config.h

```diff
@@ -29,13 +29,29 @@
 #define RING_LIGHT_LEDS 24
 #define LIGHT_BAR_LEDS 16
 
+// tentative names for LED ring modes
+typedef enum {
+  RING_LIGHT_MODE_PLACEHOLDER1, // single colour
+  SPIN,
+  RING_LIGHT_MODE_PLACEHOLDER2, // colour shift
+  RING_LIGHT_MODE_PLACEHOLDER3, // static rainbow
+  RING_LIGHT_MODE_PLACEHOLDER4, // reactive rainbow
+  REACT_TO_SCR,
+  RING_LIGHT_MODE_PLACEHOLDER5, // breathing
+  HID,
+  RING_LIGHT_MODE_ENUMS
+} ring_light_mode;
+
 typedef struct {
   uint8_t version;
   uint8_t reverse_tt;
+  ring_light_mode tt_effect;
 } config;
 
 void config_init(config* self);
 void toggle_reverse_tt(config* self);
+void cycle_tt_effects(config* self);
 
 // button combos
 #define REVERSE_TT_COMBO (BUTTON_1 | BUTTON_7 | BUTTON_8)
+#define TT_EFFECTS_COMBO (BUTTON_2 | BUTTON_8 | BUTTON_11)
```

Comment:
Many older iidx controllers only have 9 buttons, to accomodate this our button combinations should not use `BUTTON_10`/`BUTTON_11`

Reply:
Are the start buttons supposed to be wired up to 8 and 9? I thought they would be wired up to 8 and 11 based off of the layout of the headers.

Reply:
misunderstanding on my part. 8 and 11 are good, we want to avoid 9 and 10